### PR TITLE
refactor(core/pipeline): Convert a few execution source details to react

### PR DESCRIPTION
**Pull request by jrsquared**

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/config/stages/checkPreconditions/CheckPreconditionsExecutionDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { robotToHuman } from 'core/presentation/robotToHumanFilter/robotToHuman.filter';
+
+export function CheckPreconditionsExecutionDetails(props: IExecutionDetailsSectionProps) {
+  const context = get(props.stage, 'context.context', {} as any);
+  return (
+    <ExecutionDetailsSection name={props.name} current={props.current}>
+      <div className="row">
+        <div className="col-md-12">
+          <dl className="dl-horizontal">
+            {Object.keys(context).filter((key) => key !== 'expression' && context[key] !== null).map((key) => (
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/checkPreconditions/CheckPreconditionsExecutionDetails.tsx]: might be worth `.sort`ing the keys (might not be, what do I know)

File: app/scripts/modules/core/src/pipeline/config/stages/checkPreconditions/CheckPreconditionsExecutionDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+              <div key={key}>
+                <dt>{robotToHuman(key)}</dt>
+                <dd>{JSON.stringify(context[key])}</dd>
+              </div>
+            ))}
+            <div>
+              <dt>Fail Pipeline</dt>
+              <dd>{props.stage.context.failPipeline ? 'true' : 'false'}</dd>
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/stages/checkPreconditions/CheckPreconditionsExecutionDetails.tsx]: `<dd>{!!props.stage.context.failPipeline}}</dd>` ?

**Comment by anotherchrisberry** [comment]
Nice, I really like this approach

File: app/scripts/modules/core/src/pipeline/config/stages/checkPreconditions/CheckPreconditionsExecutionDetails.tsx (hunk 2 above)
Comment (jrsquared) [reply app/scripts/modules/core/src/pipeline/config/stages/checkPreconditions/CheckPreconditionsExecutionDetails.tsx]: One would hope. Alas, JSX does not stringify the Boolean value. I could do `String(!!props.stage.context.failPipeline)` though...

**Pull request merged**